WordPress/gutenberg
# Font Library: consolidate existing API rest endpoints.

## What?
- Rename font library endpoints.
- Disables some default API endpoints created when registering the wp_font_family post type.

With this PR, we reduce the number of Font Library-related endpoints.

Old paths:
```
GET /fonts/collections/
GET /fonts/collections/<id>

POST /fonts/
DELETE /fonts/

GET /wp/v2/wp_font_family
POST /wp/v2/wp_font_family

GET /wp/v2/wp_font_family/(?P<id>[\d]+)
POST /wp/v2/wp_font_family/(?P<id>[\d]+)
DELETE /wp/v2/wp_font_family/(?P<id>[\d]+)

GET /wp/v2/wp_font_family/(?P<id>[\d]+)/autosaves
POST  /wp/v2/wp_font_family/(?P<id>[\d]+)/autosaves
GET /wp/v2/wp_font_family/(?P<parent>[\d]+)/autosaves/(?P<id>[\d]+)

```

New paths:
```
GET /font-collections/
GET /font-collections/<id>

GET /font-families/
POST /font-families/
DELETE /font-families/
```


## Why?
- Remove endpoints not needed.
- Avoid exposing default endpoints with no sanitization process in place.

## How?
- Split the Font Library rest controller into 2 different controllers Font Families and Font Collections.
- Make Font Families Rest Controller extend WP_REST_Posts_Controller class.
- Disable /autosave/ endpoints by creating an empty autosave controller class.

## Testing Instructions
- Use the font library from the editor and verify all the functionality is working as expected.
- Run PHP unit tests.


---

This seems to be a better alternative than: https://github.com/WordPress/gutenberg/pull/57115
